packages/tailwindcss/src/compat/selector-parser.test.ts

```diff
@@ -131,6 +131,34 @@ describe('parse', () => {
       },
     ])
   })
+
+  it('parses nesting selector before attribute selector', () => {
+    expect(parse('&[data-foo]')).toEqual([
+      { kind: 'selector', value: '&' },
+      { kind: 'selector', value: '[data-foo]' },
+    ])
+  })
+
+  it('parses nesting selector after an attribute selector', () => {
+    expect(parse('[data-foo]&')).toEqual([
```

Comment:
While converting from the `postcss-selector-parser` setup to our own `SelectorParser`, I noticed that this case wasn't covered properly and broke some tests.